js/adapt-contrib-pageLevelProgress.js

```diff
@@ -1,66 +1,137 @@
 define([
     'core/js/adapt',
     './completionCalculations',
-    './PageLevelProgressMenuView',
-    './PageLevelProgressNavigationView'
-], function(Adapt, completionCalculations, PageLevelProgressMenuView, PageLevelProgressNavigationView) {
-
-    function setupPageLevelProgress(pageModel, enabledProgressComponents) {
-        new PageLevelProgressNavigationView({model: pageModel, collection: new Backbone.Collection(enabledProgressComponents)});
-    }
-
-    // This should add/update progress on menuView
-    Adapt.on('menuView:postRender', function(view) {
-        if (view.model.get('_id') == Adapt.location._currentId) return;
-
-        var coursePLPConfig = Adapt.course.get('_pageLevelProgress');
-
-        // do not proceed if pageLevelProgress is not enabled in course.json
-        if (!coursePLPConfig || !coursePLPConfig._isEnabled) {
-            return;
+    './PageLevelProgressNavigationView',
+    './PageLevelProgressIndicatorView',
+    './PageLevelProgressCollection'
+], function(Adapt, completionCalculations, PageLevelProgressNavigationView, PageLevelProgressIndicatorView, PageLevelProgressCollection) {
+
+    var PageLevelProgress = Backbone.Controller.extend({
+
+        initialize: function() {
+            this.listenTo(Adapt, 'app:dataReady', this.onDataReady);
+        },
+
+        getCourseConfig: function() {
+            return Adapt.course.get('_pageLevelProgress');
+        },
+
+        onDataReady: function() {
+            // Do not proceed if pageLevelProgress is not enabled in course.json
+            var coursePLPConfig = this.getCourseConfig();
+            if (!coursePLPConfig || !coursePLPConfig._isEnabled) {
+                return;
+            }
+            this.setUpEventListeners();
+        },
+
+        setUpEventListeners: function() {
+            var headerIndicatorTypes = [
+                    'menu',
+                    'page',
+                    'article',
+                    'block',
+                    'component'
+                ];
+
+            var headerIndicatorEventNames = headerIndicatorTypes
+                .concat(['']).join('View:render ');
+
+            this.listenTo(Adapt, headerIndicatorEventNames, this.renderHeaderIndicatorView);
+
+            this.listenTo(Adapt, {
+                'menuView:postRender': this.renderMenuItemIndicatorView,
+                'router:page': this.renderNavigationView
+            });
+        },
+
+        renderHeaderIndicatorView: function(view) {
+            var model = view.model;
+
+            var config = model.get('_pageLevelProgress');
+            if (!config || !config._isEnabled || !config._isCompletionIndicatorEnabled) {
+                return;
+            }
+
+            var pageModel = _.find(model.getAncestorModels(), function (item) {
```

Comment:
```js
var pageModel = model.findAncestor('pages');
```

Reply:
actually it would need to be `model.findAncestor('contentObjects');`